accepted/2022/simplify-rid-model.md

```diff
@@ -0,0 +1,221 @@
+| win-x64    | All Windows versions, for x64. |
+| win-x86    | All Windows versions, for x86. |
+
+Note: Singles are for processor-agnostic managed code.
+
+Note: All RIDs are subject to .NET support policy. For example, .NET 7 doesn't support Ubuntu 16.04. The `linux-x64` RID doesn't include that specificity.
+
+Note: `osx` is used instead of `macOS` within the RID scheme. This design change may be a good opportunity to introduce `macOS`. It probably makes sense only to do that for Arm64.
+
+The following table describes RIDs supported only by `runtimes.json`. It is not exhaustive.
+
+| RID        | Description of scope |
+|------------|----------------------|
+| any        | The root RID, which is compatible with all other RIDs. |
+| alpine     | All Alpine versions and architecture builds. |
+| alpine-x64 | All Alpine versions, for x64. Other processor-types are also supported. |
+| alpine3.16-arm64 | Alpine 3.16, for Arm64. Other processor-types are also supported.|
+| centos     | CentOS has a similar scheme as Alpine. |
+| debian     | Debian has a similar scheme as Alpine. |
+| osx        | macOS has a similar scheme as Alpine. |
+| rhel       | Red Hat Enterprise Linux has a similar scheme as Alpine. |
+| tizen      | Tizen has a similar scheme as Alpine. |
+| ubuntu     | Ubuntu has a similar scheme as Alpine. |
+
+Note: Many other Linux distros are represented in `runtimes.json`.
+
+Note: `runtimes.json` will be frozen, which means that RID schemes only supported by this file are abandoned. For example, RIDs that include OS versions will not be updated going forward.
+
+## Source-build
+
+A major design tenet of the RID system is maximizing portability of code. This is particularly true for Linux. That makes sense from the perspective of Microsoft wanting to make one build of .NET available across many Linux distros, separately for both glibc and musl. It makes less sense for distros themselves building .NET from source and publishing binaries to a distro package feed.
+
+We sometimes refer to `linux-x64` (and equally to `linux-arm64` and `linux-x86`) as the "portable Linux RID". As mentioned in the libc section, this RID establishes a wide glibc compatibility range. In addition, the RID also establishes broad compatibility across distros (and their associated versions) for .NET dependencies, like OpenSSL and ICU. The way this particular form of compatibility shows up is observable but is nuanced (and isn't explained here).
+
+We'll use Red Hat as an example of an organization that builds .NET from source, to complete this discussion. They have raised concerns on [source-build not playing nicely with `linux-x64`](https://github.com/dotnet/runtime/pull/62942#issuecomment-1056942235).
+
+For Red Hat, it makes sense to accept and support `linux-x64` assets, but not to produce or offer them. Instead, Red Hat would want to produce `rhel-x64` assets. It's easiest to describe this design-point in terms of concrete scenarios.
```

Comment:
The rid chosen by a source-build maintainer should not be the same for distribution versions that are not binary compatible. That means source-build rids are likely to have versions in them.

Can you use `rhel.8-x64` instead of `rhel-x64` to make this visible?

Reply:
Yes. I realized this after publishing the doc. I'll update this.

Reply:
Ughh. I added a mess of an answer. Thoughts?

https://github.com/dotnet/designs/pull/260/commits/840970625f490356d6de38aceea67320987f4fa9

Reply:
lgtm.

> However, for that to work, we would need to add `rhel.8-x64` to `runtimes.json`. Otherwise, NuGet won't be able to realize that `rhel.8-x64` should be treated as `linux-x64` for restoring NuGet assets. However, we want to freeze `runtimes.json`. That topic is left as an unresolved issue in this spec, at least for now.

dotnet/runtime has support for patching the `runtime.json` graph with the source-build rid. See https://github.com/dotnet/runtime/pull/50818 by @ericstj.

Reply:
Yup. You are right. Will resolve that. Eric separately gave me the same context.

Reply:
Added to the doc.